### PR TITLE
commit testcases to run in docker host environment

### PR DESCRIPTION
**Pull request by tharindu1st**

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.core/pom.xml [hunk 1]
```diff
@@ -578,6 +578,9 @@
                                             <ACCEPT_EULA>Y</ACCEPT_EULA>
                                             <SA_PASSWORD>wso2apim123#</SA_PASSWORD>
                                         </env>
+                                        <ports>
+                                            <port>11433:1433</port>
+                                        </ports>
```
Comment (pubudu538) [inline components/apimgt/org.wso2.carbon.apimgt.core/pom.xml]: Will those ports assign dynamically? If not shall we try to that?

File: components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/TestUtil.java [hunk 2]
```diff
@@ -148,16 +144,20 @@ public static Application createSummaryApplication(Application app){
...
-        if (ipAddressMap.containsKey(aliasName)){
-            return ipAddressMap.get(aliasName);
-        }else{
-            String ipAddress = docker.inspectContainer(aliasName).networkSettings().ipAddress();
-            ipAddressMap.put(aliasName,ipAddress);
-            return ipAddress;
+
+    /**
+     * Utility for get Docker running host
+     * @return
```
Comment (anuruddhal) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/TestUtil.java]: Add the return description

File: components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/TestUtil.java [hunk 3]
```diff
@@ -148,16 +144,20 @@ public static Application createSummaryApplication(Application app){
...
+     * @throws URISyntaxException
+     */
+    public String getIpAddressOfContainer() throws URISyntaxException {
+        String ip = "localhost";
+        String dockerHost = System.getenv("DOCKER_HOST");
+        if (!StringUtils.isEmpty("DOCKER_HOST: " + dockerHost)) {
```
Comment (pubudu538) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/TestUtil.java]: Shall we add default value for the DOCKER_HOST as localhost

File: components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/TestUtil.java [hunk 4]
```diff
@@ -148,16 +144,20 @@ public static Application createSummaryApplication(Application app){
...
-            return ipAddressMap.get(aliasName);
-        }else{
-            String ipAddress = docker.inspectContainer(aliasName).networkSettings().ipAddress();
-            ipAddressMap.put(aliasName,ipAddress);
-            return ipAddress;
+
+    /**
+     * Utility for get Docker running host
+     * @return
+     * @throws URISyntaxException
```
Comment (anuruddhal) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/TestUtil.java]: Add exception details.

File: components/apimgt/org.wso2.carbon.apimgt.core/pom.xml (hunk 1 above)
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.core/pom.xml]: these values we can add dynamically,
If we add we can't gurantee the port range.

File: components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/TestUtil.java (hunk 3 above)
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.core/src/test/java/org/wso2/carbon/apimgt/core/TestUtil.java]: we couldn't set this value as localhost for docker host contains tcp://localhost:2375 like tcp address

File: components/apimgt/org.wso2.carbon.apimgt.core/pom.xml (hunk 1 above)
Comment (tharindu1st) [inline components/apimgt/org.wso2.carbon.apimgt.core/pom.xml]: when build team assign the port range, We can use those ranged port for every port

Comment (pubudu538) [inline components/apimgt/org.wso2.carbon.apimgt.core/pom.xml]: Got it. Thank you!

**Pull request merged**